experiments/download_models.sh

```diff
@@ -3,5 +3,52 @@
 # Always sync to ../data, relative to this script.
 SCRIPT_DIR="$(cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd)"
 PROJECT_DIR="$(dirname ${SCRIPT_DIR})"
+DATA_DIR="${PROJECT_DIR}/data"
 
-aws s3 sync s3://shwang-chai/public/data/ ${PROJECT_DIR}/data
+DRY_RUN_MODE=false
+ALL_MODE=false
+TEMP=$(getopt -o '' -l all,dryrun -- $@)
+
+if [[ $? != 0 ]]; then exit 1; fi
+eval set -- "$TEMP"
+
+while true; do
+  case "$1" in
+    --dryrun)
+      DRY_RUN_MODE=true
+      shift
+      ;;
+    --all)
+      # Download all data (by default, skips large meta-data/log files).
+      # As of 2019.Nov.12, the difference in download size was 77MB vs ~800MB.
+      ALL_MODE=true
+      shift
+      ;;
+    --)
+      shift
+      break
+      ;;
+    *)
+      echo "Unrecognized flag $1" >&2
+      exit 1
+      ;;
+  esac
+done
+
+FLAGS=""
+
+if not $ALL_MODE; then
+  FLAGS+="--exclude '*/monitor/*' \
+    --exclude '*/parallel/*' \
+    --exclude '*/sacred/*' \
+    --exclude '*events.out.tfevents*' "
+fi
+
+if $DRY_RUN_MODE; then
+  FLAGS+='--dryrun '
+elif [[ -d ${DATA_DIR} ]]; then
+  rm -r ${DATA_DIR}
+fi
+
+command="aws s3 sync ${FLAGS} s3://shwang-chai/public/data/ ${DATA_DIR}"
+eval $command  # Required for correct interpretation of ALL_MODE flags.
```

Comment:
I'm surprised this is needed. What were you trying before and what error did you see?

Reply:
The exclude flags don't seem to work properly without this set up. I'm guessing this is because they are all lumped together in one string and the `aws` parser somehow gets confused?

If I run this command directly, then I `aws` starts downloading excluded files like the monitor data:
```
(dryrun) download: s3://shwang-chai/public/data/expert_models/acrobot_0/monitor/mon000.monitor.csv to data/expert_models/acrobot_0/monitor/mon000.monitor.csv
(dryrun) download: s3://shwang-chai/public/data/expert_models/acrobot_0/monitor/mon001.monitor.csv to data/expert_models/acrobot_0/monitor/mon001.monitor.csv
(dryrun) download: s3://shwang-chai/public/data/expert_models/acrobot_0/monitor/mon002.monitor.csv to data/expert_models/acrobot_0/monitor/mon002.monitor.csv
```

Reply:
I've made a commit that fixes this using `bash` arrays.

I'd generally avoid `eval` unless you need to use things like redirection, pipes. It just splits on whitespace, so will break if you ever had e.g. a directory with whitespace in it that you wanted to exclude.

Reply:
Thanks!